File: src/main/java/com/rabit/api/dto/response/CompanyResponse.java
package com.rabit.api.dto.response;

import com.rabit.api.model.enums.CompanyRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CompanyResponse {

    private UUID companyId;

    private String name;

    private String urlSlug;

    private String description;

    private String addressLine1;

    private String addressLine2;

    private String city;

    private String state;

    private String country;

    private String pinCode;

    private String contactNumber;

    private boolean hasLogo;

    private String logoUrl;

    private UUID createdBy;

    private Instant createdAt;

    private Instant updatedAt;

    private boolean active;

    private CompanyRole userRole;

    private String message;
}

File: dtos/responses/CompanyResponse.js
class CompanyResponse {
  constructor(company, userRole) {
    this.companyId = company.companyId;
    this.name = company.name;
    this.urlSlug = company.urlSlug;
    this.description = company.description || null;
    this.addressLine1 = company.addressLine1;
    this.addressLine2 = company.addressLine2 || null;
    this.city = company.city;
    this.state = company.state;
    this.country = company.country;
    this.pinCode = company.pinCode;
    this.contactNumber = company.contactNumber;
    this.hasLogo = company.logo !== null && company.logo !== undefined;
    this.createdBy = company.createdBy || null;
    this.createdAt = company.createdAt;
    this.updatedAt = company.updatedAt;
    this.active = company.active;
    this.userRole = userRole ? { roleId: userRole.roleId, name: userRole.name } : null;
  }
}
module.exports = CompanyResponse;
